wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -334,10 +325,18 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
           logger.debug(
             s"Marked utxo=${updated.toHumanReadableString} as state=${updated.state}"))
         updatedF.map(Some(_))
-      case TxoState.Reserved | TxoState.PendingConfirmationsSpent |
-          BroadcastSpent =>
+      case state @ (TxoState.Reserved | TxoState.PendingConfirmationsSpent |
```

Comment:
This state transistion is probably correct for `TxoState.{Reserived,BroadcastSpent}` but more thought needs to be put into `PendingConfirmationsSpent`. 

What this logic is effectively saying is we are reorg the existing `spendingTxId` that HAS confirmations on it (`PendingConfirmationsSpent`) for a new spending txid that has zero confs. 

Is this what we want?